pennylane/resource/error.py

```diff
@@ -77,3 +78,43 @@ def error(self) -> AlgorithmicError:
         Returns:
             AlgorithmicError: The error.
         """
+
+
+class SpectralNormError(AlgorithmicError):
+    """Class representing the spectral norm error.
+
+    The spectral norm error is defined as the distance (in spectral norm) between the true unitary
+    we wish to apply and the approximate unitary that we actually apply.
+
+    Args:
+        error (float): The numerical value of the error
+
```

Comment:
It'd be great to have an example here of how a user would interact with this. @DSGuala and @Jaybsoni could help with thinking about what we put here.

Reply:
+1, having these docs be high quality can really help the impact of the work we are doing here.

Reply:
I would also recommend examples in the `combine()` and `get_error()` methods.

Reply:
Added examples for the class and its methods. Note that the class and `combine()` examples are the same.

Reply:
Thank you, @soranjh!

Reply:
Thanks!